logprep/ng/processor/list_comparison/processor.py

```diff
@@ -81,14 +81,20 @@ def _list_comparison(self, rule: ListComparisonRule, event: dict) -> tuple[list[
         the result (key) and a list of filenames pertaining to said result (value).
         """
 
+        # get value that should be checked in the lists
         field_value = get_dotted_field_value(event, rule.source_fields[0])
-
-        list_matches = [
-            compare_list
-            for compare_list in rule.compare_sets
-            if field_value in rule.compare_sets[compare_list]
-        ]
-
-        if not list_matches:
+        value_list = field_value if isinstance(field_value, list) else [field_value]
+
+        # iterate over lists and check if element is in any
+        list_matches = []
+        for field_value in value_list:
+            for compare_list in rule.compare_sets:
+                if compare_list in list_matches:
+                    continue
+                if field_value in rule.compare_sets[compare_list]:
+                    list_matches.append(compare_list)
```

Comment:
Optional:
One way to simplify this could be to convert everything to sets and simply do intersections checks. However, intersections compute the whole intersection and don't terminate after the first overlap, [so it would probably be less performant](https://stackoverflow.com/questions/6821329/speed-differences-between-intersection-and-object-for-object-in-set-if-object).

Reply:
Thank you for the suggestion, In my opinion performance is more important in this place, even if it's just a small difference.